app/src/main/java/com/github/ybecker/epforuml/notifications/PushNotificationService.kt

```diff
@@ -0,0 +1,105 @@
+package com.github.ybecker.epforuml.notifications
+
+import android.Manifest
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.ContentValues
+import android.content.ContentValues.TAG
+import android.content.Context
+import android.content.pm.PackageManager
+import android.util.Log
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.android.volley.Request
+import com.android.volley.Response
+import com.android.volley.toolbox.JsonObjectRequest
+import com.android.volley.toolbox.Volley
+import com.github.ybecker.epforuml.R
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import org.json.JSONObject
+
+open class PushNotificationService: FirebaseMessagingService() {
```

Comment:
Maybe it would be safer to implement onNewToken as a change of token might cause some issues so it would be better to treat this case in this method.